src/sentry/replays/lib/storage.py

```diff
@@ -116,7 +117,11 @@ def get(self, segment: RecordingSegmentStorageMeta) -> bytes:
     @metrics.wraps("replays.lib.storage.StorageBlob.set")
     def set(self, segment: RecordingSegmentStorageMeta, value: bytes) -> None:
         storage = get_storage(self._make_storage_options())
-        storage.save(self.make_key(segment), BytesIO(value))
+        try:
+            storage.save(self.make_key(segment), BytesIO(value))
+        except TooManyRequests:
+            # if we 429 because of a dupe segment problem, ignore it
```

Comment:
is the only reason we can 429 a dupe problem? Can we get ourselves in a situation where we're dropping data due to this and have no visibility due to no logging/metrics?

Reply:
looking at this https://cloud.google.com/storage/quotas it seems unlikely we'd hit other quotas. but i can add a metric for visibility.